boxoffice/models/payment.py

```diff
@@ -0,0 +1,89 @@
+from datetime import datetime
+import decimal
+from coaster.utils import LabeledEnum
+from baseframe import __
+from boxoffice.models import db, BaseMixin, Order
+
+__all__ = ['OnlinePayment', 'PaymentTransaction', 'PaymentFailError']
+
+
+class RAZORPAY_PAYMENT_STATUS(LabeledEnum):
+    """
+    Reflects payment statuses as specified in
+    https://docs.razorpay.com/docs/return-objects
+    """
+    CREATED = (0, __("Created"))
+    AUTHORIZED = (1, __("Authorized"))
+    CAPTURED = (2, __("Captured"))
+    # Only fully refunded payments.
+    REFUNDED = (3, __("Refunded"))
+    FAILED = (4, __("Failed"))
+
+
+class OnlinePayment(BaseMixin, db.Model):
+    """
+    Represents payments made through a payment gateway.
+    Supports Razorpay only.
+    """
+    __tablename__ = 'online_payment'
+    __uuid_primary_key__ = True
+    customer_order_id = db.Column(None, db.ForeignKey('customer_order.id'), nullable=False)
+    order = db.relationship(Order, backref=db.backref('online_payments', cascade='all, delete-orphan'))
+
+    # Payment id issued by the payment gateway
+    pg_payment_id = db.Column(db.Unicode(80), nullable=False)
```

Comment:
Don't use the `_id` suffix if it's not a foreign key. It's confusing when seeing references elsewhere in code.